Weather app tweaks

NotherNgineer opened the pull request:
Proposed minor changes to weather app.

1.  Allow frequency to be saved across runs in App Settings file.
2.  Use bandwidth, sample rate, and modulation set in RadioState constructor (and allow override in App Settings file).
3.  Propose changing frequency step to a larger value (0.1 Mhz).
4.  Changed icon color to represent new development state.  See: https://github.com/eried/portapack-mayhem/wiki/Applications
5.  Added support for displaying temperatures in Fahrenheit, and save units preference in App Settings file.

Feedback requested from @htotoo on these proposed changes.

gullradriel reviewed (approved):
Not sure @htotoo can review.
Looks good to me, thought I would only downgrade to yellow as it's working and only miss a few features

NotherNgineer commented:
@gullradriel  I think the color is more about setting user expectations (I wouldn't call it a "downgrade") but I've taken your suggestion and changed it to yellow.

I have also added support for displaying temperatures in F, and to save this preference in App Settings.
